app/nagbot.py

```diff
@@ -128,13 +151,35 @@ def execute_internal(self, channel):
         else:
             sqslack.send_message(channel, 'No instances were stopped today.')
 
-
     def execute(self, channel):
         try:
-            self.execute_internal(channel)
+            self._execute_internal(channel)
         except Exception as e:
             sqslack.send_message(channel, "Nagbot failed to run the 'execute' command: " + str(e))
-            raise(e)
+            raise (e)
+
+
+def get_taggable_instances(instances):
+    return list(i for i in instances if is_taggable(i))
+
+
+def is_taggable(instance):
+    return (not is_whitelisted(instance) and
+            (not instance.stop_after
+             or not instance.terminate_after
+             or (not parsing.parse_date_tag(instance.terminate_after).expiry_date
+                 and not parsing.parse_date_tag(instance.terminate_after).warning_date)))
+
+
+def tag_instance(instance):
+    if not instance.stop_after:
+        sqaws.set_tag(instance.region_name, instance.instance_id, 'Stop after',
+                      AUTO_STOP_AFTER_DAY_YYYY_MM_DD)
+    if (not instance.terminate_after
```

Comment:
This would be a bit more readable if we pulled out a helper method e.g. `_terminate_after_date_present(instance)` or something; could also be used by `is_taggable()`

Reply:
Actually, it could be a method on `Instance`

Reply:
I moved `is_stop_after_tag_missing` and `is_terminate_after_tag_missing` to `Instance`. I chose not to pass the `whitelist` logic into `Instance` or move other functions like `is_stoppable`, which would require moving or passing the DateTime constants for now. I would like to do so in the future, but maybe in more of a 0.0.1 pass.